sklearn/preprocessing/data.py

```diff
@@ -1538,10 +1540,65 @@ def transform(self, X):
                         columns.append(bias)
                 XP = sparse.hstack(columns, dtype=X.dtype).tocsc()
             else:
-                XP = np.empty((n_samples, self.n_output_features_),
-                              dtype=X.dtype, order=self.order)
-                for i, comb in enumerate(combinations):
-                    XP[:, i] = X[:, comb].prod(1)
+                # When the matrix is small, it is faster to transpose
+                # the matrix first to multiply contiguous memory segments.
+                transpose = X.size <= 1e7 and self.order == 'C'
+                n = X.shape[1]
+
```

Comment:
When the size is larger, is transposing slower or equivalent ?
And, when the matrix is small does the transposing bring a significant part of the speed up ?

My point is that the transpose switch makes the code twice as long, and I'm wondering if we could always do one or the other.

Reply:
![image](https://user-images.githubusercontent.com/4695832/53563525-96837500-3b54-11e9-84d3-4068d82f2e98.png)

I remove the transpose and it seemed better. So i'll keep it. I added it because it was better before I removed one unnecessary copy this morning. transpose is inefficient when order=='F' so the only option to simplify is to remove it. Updating the PR.